support_branding/static/src/js/support_branding.js

```diff
@@ -2,105 +2,169 @@
-                                },
-                            ])
-                            .then(
-                                function (mail_id) {
-                                    return mail_mail.call("send", [[mail_id]]);
-                                },
-                                function () {
-                                    // If the call failed, fire the mailto link
-                                    // hoping there is a properly configured email
-                                    // client
-                                    $body.val($description.val() + "\n" + $body.val());
-                                    $button.unbind("click");
-                                    $button.click();
-                                }
-                            )
-                            .then(function () {
-                                $form.parents(".modal").modal("hide");
-                            });
                         ev.preventDefault();
+                        var desc = $description.val();
+                        var params = {
+                            state: "outgoing",
+                            auto_delete: true,
+                            email_to: self.support_cp_email,
+                            subject: subject,
+                            body_html: jQuery("<div/>")
+                                .append(
+                                    jQuery("<div/>").text(desc),
+                                    jQuery("<pre/>").text(body)
+                                )
+                                .html(),
+                        };
+                        self._rpc({
+                            model: "mail.mail",
+                            method: "create",
+                            args: [params],
+                        }).then(
+                            function (mail_id) {
+                                if (mail_id) {
+                                    self._rpc({
+                                        model: "mail.mail",
+                                        method: "send",
+                                        args: [mail_id],
+                                    }).then(function (res) {
+                                        if (res) {
+                                            self.do_notify(
+                                                "Success",
+                                                "Support mail created!"
+                                            );
+                                        }
+                                    });
+                                }
+                            },
+                            function () {
+                                $body.val(desc + "\n" + $body.val());
+                                $btn.unbind("click");
+                                $btn.click();
+                            }
+                        );
```

Comment:
Usability here is not optimal:

![image](https://user-images.githubusercontent.com/1466356/126187425-7437ee53-c16e-4b86-a6da-ff44447b8326.png)

1. We are seeing an "OK" button and an "Email to **company**" button. Clicking "OK" closes the dialog without sending an email, which is not what a user would expect.
2. Clicking the "Email to **company**" button does send the email, but silently, there is no feedback and the dialog stays open. This will confuse users and let them press many times, so multiple mails being sent.

Reply:
this has been restructured and notification sent to a user, disable OK button since we already have close on right top of dialog, and also after successfully sending, dialog will close.